### PR TITLE
DAOS-10603 control: Add disable_hugepages server config file parameter

### PR DESCRIPTION
**Pull request by tanabarr**
Replace setting nr_hugepages: -1 with an explicit disable_hugepages: true
directive to improve clarity of intent.

Features: control

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by tanabarr** [comment]
@knard-intel is there anywhere else I need to update regarding the change from using `nr_hugepages: -1` to `disable_hugepages: true` that you are aware of? Specifically thinking of any docker usages.

**Review by mjmac — DISMISSED** [review]

**Comment by knard-intel** [comment]
> > @knard-intel is there anywhere else I need to update regarding the change from using `nr_hugepages: -1` to `disable_hugepages: true` that you are aware of? Specifically thinking of any docker usages.
> 
> I am going to investigate on it... but at first glance I have not used this feature with the docker stuff that I have pushed. I keep you inform asap.

For the docker stuff that I pushed the `nr_hugepages` must be greater to 0 for working properly, and thus is not impacted by this patch.
However (from what I know), the functional tests done with a virtual machine do not use NVMe disks and could be impacted by this modification. I will check tomorrow the daos_server.yml generated for such tests.

**Review by kjacque — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

File: src/control/server/config/server.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ type Server struct {
 	DisableVMD          bool                   `yaml:"disable_vmd"`
 	EnableHotplug       bool                   `yaml:"enable_hotplug"`
 	NrHugepages         int                    `yaml:"nr_hugepages"` // total for all engines
```
Comment (mjmac) [inline src/control/server/config/server.go:51]: Should this be changed to a uint? A little concerned that existing configs may have the -1 value and I'm not sure what happens in that case.

Comment (tanabarr) [reply src/control/server/config/server.go:51]: I started changing this to uint but the cascading updates made me think it wasn't a worthwhile change, you end up having to cast things like uint(len(<slice>)) and that doesn't feel nice. We validate that the value is not <0 early in the flow and tbh overall feels better without changing to uint. happy to change back if you feel like it though.

**Review by kjacque — APPROVED** [review]

File: utils/config/daos_server.yml [hunk 2]
```diff
@@ -200,11 +200,15 @@
 #
 ## Example: (2 engines * (16 targets/engine * 1GiB)) / 2MiB hugepage size = 16834
 #
+## default: 0
```
Comment (knard-intel) [inline utils/config/daos_server.yml:203]: Maybe some comments on this parameter are still needed to inform the enduser of its role and its allowed values.

**Review by knard-intel — APPROVED** [review]
Just one minor comments on the configuration file template.
I have also checked the server configuration file used for VM functional tests and the nr_hugepages parameters should be compliant with this patch :)
Example of such configuration file could be found at 
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-9203/2/artifact/Functional%20on%20EL%208/pool/destroy.py/daos_configs/daos_server.wolf-103vm2.yml/*view*/

File: utils/config/daos_server.yml (hunk 2 above)
Comment (tanabarr) [reply utils/config/daos_server.yml:203]: @Michael-Hennecke recently updated these comments regarding nr_hugepages. will leave him to decide if they should be added to.

Comment (knard-intel) [reply utils/config/daos_server.yml:203]: OK

**Review by kjacque — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Pull request merged**